pkg/util/kubernetes/apiserver/apiserver.go

```diff
@@ -206,7 +206,8 @@ func WaitForAPIClient(ctx context.Context) (*APIClient, error) {
 	}
 }
 
-func getClientConfig(timeout time.Duration, qps float32, burst int) (*rest.Config, error) {
+// GetClientConfig returns a REST client configuration
+func GetClientConfig(timeout time.Duration, qps float32, burst int) (*rest.Config, error) {
```

Comment:
why do we need this function to be public?

Reply:
That’s because the KSM CustomResource discoverer takes a client config here: https://github.com/DataDog/datadog-agent/pull/31715/files#diff-b2d421ac10f97e5f34823072a47feadbe0b52f67f17b5baa9744a7b8ffe8fddcR126
I wanted the client used by KSM to respect the settings that the agent uses.